torch/_custom_op/autograd.py

```diff
@@ -0,0 +1,222 @@
+import torch
+import torch.utils._pytree as pytree
+from collections import namedtuple
+
+
+# NOTE [CustomOp autograd kernel indirection]
+def autograd_kernel_indirection(custom_op):
+    # TODO(#101191): Use the actual C++ autograd not implemented fallback,
+    # or change the default autograd fallback to the autograd not implemented fallback.
+    def autograd_not_implemented(*args, **kwargs) -> None:
+        if pytree.tree_any(
+            lambda x: isinstance(x, torch.Tensor) and x.requires_grad, (args, kwargs)
```

Comment:
Yes, good point

Reply:
It is a good idea to give people an easy way to do this (so they don't need to define their save_for_backward function and their backward function in order to say that their function is not differentiable). I can address that as a follow up, unless you want to see it in this PR?

I'm not really sure what this should look like yet: we could ask the author to give us a `differentiable` flag at creation of the custom_op, or we could ask them to `impl_backward(None)`, or something else.

Reply:
Follow up PR sounds ok!

> I'm not really sure what this should look like yet: we could ask the author to give us a differentiable flag at creation of the custom_op, or we could ask them to impl_backward(None), or something else.

My vote would be for having a `differentiable` flag at custom_op creation since whether the op is differentiable or not seems like a property of the op itself.